src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,187 @@
+package org.corfudb.infrastructure;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.runtime.view.LayoutView;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+
+/**
+ * Instantiates and performs failure detection adn handling asynchronously.
+ * <p>
+ * Created by zlokhandwala on 9/28/16.
+ */
+@Slf4j
+public class ManagementServer extends AbstractServer {
+
+    @Override
+    public void reset() {
+    }
+
+    /**
+     * The options map.
+     */
+    Map<String, Object> opts;
+
+    private ServerContext serverContext;
+
+    private IFailureDetectorPolicy failureDetectorPolicy;
+    private LayoutView layoutView = null;
+    private CorfuRuntime corfuRuntime = null;
+
+    // Stores this node's address
+    private String localEndpoint;
+
+    // Used for initial bootstrap only.
+    private LayoutServer layoutServer;
+
+    // Bootstrap Layout
+    private Layout bootstrapLayout = null;
+
+    // Flag to be toggled once bootstrapped.
+    private static boolean isBootstrapped = false;
+
+    /**
+     * A scheduler, which is used to schedule failure detection.
+     */
+    private final ExecutorService schedulerService =
+            Executors.newSingleThreadExecutor(
+                    new ThreadFactoryBuilder()
+                            .setDaemon(true)
+                            .setNameFormat("Config-Mgr-%d")
```

Comment:
Please use a more appropriate name.

Also, we shouldn't allocate a thread per monitored host. In other words, we want a couple of threads monitoring 1000s of hosts as opposed to 1000s of threads.